src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    aNSC_set_value_str(value, 4);
+    shop_master->value = value;
+}
+
+static void aNSC_set_pw_password_str(NPC_SHOP_MASTER_ACTOR* shop_master) {
+    mMsg_Window_c* msg_win = mMsg_Get_base_window_p();
+    u8 *msg = shop_master->password_str;
+    mMsg_Set_item_str(msg_win, mMsg_ITEM_STR3, msg, 14);
+    mMsg_Set_item_str(msg_win, mMsg_ITEM_STR4, &msg[14], 14);
+}
+
+static void aNSC_set_pw_info_str(NPC_SHOP_MASTER_ACTOR* shop_master) {
+    mMsg_Window_c* msg_win = mMsg_Get_base_window_p();
+    mMsg_Set_free_str(msg_win, mMsg_FREE_STR6, shop_master->password.str1, 8);
+    mMsg_Set_free_str(msg_win, mMsg_FREE_STR7, shop_master->password.str0, 8);
+    aNSC_set_item_name_str(shop_master->password.item, 2);
+}
+
+static int aNSC_pc_check_password_famicom(mMpswd_password_c *pswd) {
+    int res = aNSC_PSW_RES_0;
+    if (pswd->hit_rate_index == 1) {
+        if(mMpswd_check_name(pswd) == TRUE) {
+            res = aNSC_PSW_GOOD_FAMICON;
+        } else {
+            res = aNSC_PSW_BAD;
+        }
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password_npc(mMpswd_password_c *pswd) {
+    int res = aNSC_PSW_RES_0;
+    if (pswd->hit_rate_index == 1 && mMpswd_check_npc_code(pswd) == TRUE) {
+        if (mMpswd_check_name(pswd) == TRUE) {
+            res = aNSC_PSW_GOOD_NPC;
+        } else {
+            res = aNSC_PSW_BAD;
+        }
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password_card_e(mMpswd_password_c *pswd) {
+    return aNSC_PSW_CARDE;
+}
+
+static int aNSC_pc_check_password_magazine(mMpswd_password_c *pswd) {
+    static f32 hit_rate_magazine[5] = {80.0, 60.0, 30.0, 0.0, 100.0};
+    int res = 0;
+
+    if (pswd->hit_rate_index <= 4) {
+        if (RANDOM_F(100.0) < hit_rate_magazine[pswd->hit_rate_index]) {
+            res = aNSC_PSW_GOOD_MAGAZINE_1;
+        } else {
+            res = aNSC_PSW_GOOD_MAGAZINE_2;
+        }
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password_card_e_mini(mMpswd_password_c *pswd) {
+    int res = aNSC_PSW_RES_0;
+    if (pswd->hit_rate_index == 1) {
+        res = aNSC_PSW_GOOD_CARDEM;
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password_user(mMpswd_password_c *pswd) {
+    int res = aNSC_PSW_RES_0;
+    if (pswd->hit_rate_index == 1) {
+        if (mMpswd_check_name(pswd) == TRUE) {
+            res = aNSC_PSW_GOOD_USER;
+        } else {
+            res = aNSC_PSW_BAD;
+        }
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password(NPC_SHOP_MASTER_ACTOR *shop_master) {
+    static int (*pc_check_password_proc[6])(mMpswd_password_c *) = {
+        aNSC_pc_check_password_famicom,
+        aNSC_pc_check_password_npc,
+        aNSC_pc_check_password_card_e,
+        aNSC_pc_check_password_magazine,
+        aNSC_pc_check_password_card_e_mini,
+        aNSC_pc_check_password_user
+    };
+
+    int res = aNSC_PSW_RES_0; 
+    u8 buff[24];
+
+    if (shop_master->password_str[mMpswd_PASSWORD_STR_LEN - 1] == ' ') {
```

Comment:
Since AC doesn't adhere to any standard encoding format I'd prefer you use the definitions found in m_font.h. For space that'd be `CHAR_SPACE`.